examples/service_worker/src/lib.rs

```diff
@@ -0,0 +1,337 @@
+                        .into(),
+                )
+            }
+            ServiceWorkerStatus::StateChange(state) => {
+                model.message = Some(format!("Service worker state changed to: {:?}", state))
+            }
+            ServiceWorkerStatus::SubscriptionRetrieved => {
+                model.message = Some("PushManager Subscription retrieved.".into())
+            }
+        },
+        Msg::RequestNotificationPermission => {
+            if let Some(sw_reg) = model.sw_reg.clone() {
+                orders.perform_cmd(async move {
+                    match register_push_manager(sw_reg.clone()).await {
+                        Ok(msg) => msg,
+                        Err(e) => Msg::FailedToRegisterPushManager(e),
+                    }
+                });
+            }
+        }
+        Msg::ServiceWorkerActivated => {
+            model.activated = true;
+
+            let permission: web_sys::NotificationPermission = web_sys::Notification::permission();
+            if permission == web_sys::NotificationPermission::Granted {
+                model.notifications_granted = true;
+            }
+        }
+        Msg::SubscriptionRetrieved(push_subscription) => {
+            log!(
+                "Got a push subscription of:",
+                serde_json::to_string_pretty(&push_subscription).unwrap()
+            );
+            model.notifications_granted = true;
+            model.push_subscription = Some(push_subscription);
+            orders.send_msg(Msg::StatusUpdate(
+                ServiceWorkerStatus::SubscriptionRetrieved,
+            ));
+        }
+        Msg::SendMessage => {
+            web_sys::Notification::new("Hello from Seed service worker!")
+                .expect("Couldn't send notification.");
+        }
+        Msg::SetServiceWorker(sw_reg) => {
+            model.sw_reg = Some(sw_reg);
+        }
+    }
+}
+
+// ------ ------
+//     View
+// ------ ------
+
+fn view(model: &Model) -> Node<Msg> {
+    div![
+        h1!["Seed - Service Worker Demo"],
+        p!["When the page first loads, the service worker will cache all assets, but permissions for notifications must be granted."],
+        ol![
+            li![
+                "Click the ",
+                b!["Request Push Subscription"],
+                " button and when prompted, select ",
+                b!["Allow Notifications."],
+            ],
+            li!["The subscription information will be printed to the page and notifications will now be granted."],
+            li![
+                "Click the ",
+                b!["Send Message"],
+                " button to see a browser notification coming from service worker.",
+            ],
+        ],
+        button![ev(Ev::Click, |_| Msg::RequestNotificationPermission), "Request Push Subscription"],
+        button![
+            ev(Ev::Click, |_| Msg::SendMessage),
+            IF!(not(model.notifications_granted) => attrs!{At::Disabled => true}),
```

Comment:
There is a bug because of a bad Seed API.
Please add `.as_at_value()`:
```suggestion
            IF!(not(model.notifications_granted) => attrs!{At::Disabled => true.as_at_value()}),
```
- We try to eliminate this and similar problems with the new API: https://github.com/seed-rs/seed/issues/525
- See https://seed-rs.org/0.7.0/attributes - `as_at_value` for more info.

Reply:
Updated to:
```
attrs!{At::Disabled => (!model.notifications_granted).as_at_value()},
```